### ISSUE TITLE
Errors when complexity exceeds the capability of the system to render - accessibility implications

### ISSUE THREAD

**Issue by lwolberg** [w3c/imsc-hrm#23]
I am inquiring on behalf of APA WG: 

APA understands that the specification relates to subtitles rendering. 

Captions and other such synchronized streaming text is critical to many accessible user experiences, and indeed WCAG is cited in the referenced specifications TTML and TTML Profiles. 

Our question is: if the IMSC Hypothetical render model (specified here) decides that complexity exceeds the capability of the system to render, and returns an error message, should readers of this specification be alerted that this error impacts the accessibility of the system? Or are such error states already handled in the parent specifications, and this document is only a low level API without any impact on the experience of accessibility?

**Comment by palemieux** [comment]
I like the idea of adding a note highlighting that a document that fails the HRM may negatively impact accessibility of the document and its associated content -- assuming I understood the feedback correctly.

**Comment by lwolberg** [comment]
Yes, a note along the lines of, "a document that fails the HRM may negatively impact accessibility of the document and its associated content" is an accurate statement of our concern. 

We also ask, should the error message convey the significance of this failure to the system utilizing the model? 

If HRM is only utilized by TTML, then there are sufficient notices in TTML to interpret the impact of such an HRM failure. TTML states, "Authors and implementers are encouraged to consult [media-accessibility-reqs], which presents accessibility requirements users with disabilities have with respect to audio and video on the web."

However, if HRM models are used by other systems, the significance of this failure may not be clear enough. If that is the case, the note would be more effective if it added text explaining the significance of the failure e.g., "see Success Criterion 1.1.1 at [WCAG21]." 

The spec could then add the reference,  [WCAG21],
Web Content Accessibility Guidelines (WCAG) 2.1. Andrew Kirkpatrick; Joshue O Connor; Alastair Campbell; Michael Cooper. W3C. 5 June 2018. W3C Recommendation. URL: https://www.w3.org/TR/2018/REC-WCAG21-20180605/

**Comment by nigelmegitt** [comment]
Refactoring the HRM so that it is a standalone document aside from the IMSC specifications means that it can in principle be referenced independently, so I agree it would make sense to ensure that such use possibly does warrant extra language.

I would add:
1. Informative text, e.g. a note, to say that where the HRM forms part of the document requirements for a delivery system, receiving processors might not present as authored any ISDs or documents that fail the HRM constraints, resulting in a negative impact on the accessibility of the document and its associated content.
2. Add a statement that the HRM is designed to apply alongside the constraints of IMSC, so if it is used for documents that do not adhere to those constraints, the results may not be as expected.

**Comment by palemieux** [comment]
@lwolberg Please review #24
